utils/python/CIME/SystemTests/system_tests_common.py

```diff
@@ -48,70 +51,142 @@ def __init__(self, case, expected=None):
         self._case.set_value("TEST",True)
         self._case.flush()
 
-    def fail_test(self):
-        self._runstatus = "FAIL"
+    def build(self, sharedlib_only=False, model_only=False):
+        """
+        Do NOT override this method, this method is the framework that
```

Comment:
build method is now a framework for the build phase that all subclasses must follow.